backend/controller/samples.py

```diff
@@ -11,32 +11,25 @@
 
 
 # Get the list of samples ID of the project
-def get_list(projectId, data):
+def get_list(dataProviderId, projectId, data):
     # Option 1 : get samples id list
     # Option 2 : get samples id list from selections (intersection or union)
     # Option 3 : get samples id list from model results (common or not)
     # Option 4 : mix of 2 and 3
     # Return option : from and to for streaming purpose
-
-    dataProviderId = projectId.split("|")[0]
-    projectId = projectId.split("|")[1]
-
     try:
         data_provider = data_provider_manager.get_single_data_provider(dataProviderId)
 
         # Call our utility function
-        data_id_list = get_id_list.get_list(projectId, data_provider, data)
+        data_id_list = get_id_list.get_list(data_provider, projectId, data)
```

Comment:
Interesting suggestion, because the code of the id_list util works for all data_providers, the code would be moved to the data-provider abstract class (it wouldn't be abstract anymore).

Let's discuss this later

Reply:
Sure